kubernetes/resource_kubernetes_config_map_v1_data.go

```diff
@@ -0,0 +1,213 @@
+package kubernetes
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func resourceKubernetesConfigMapV1Data() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceKubernetesConfigMapV1DataCreate,
+		ReadContext:   resourceKubernetesConfigMapV1DataRead,
+		UpdateContext: resourceKubernetesConfigMapV1DataUpdate,
+		DeleteContext: resourceKubernetesConfigMapV1DataDelete,
+		Schema: map[string]*schema.Schema{
+			"metadata": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "The name of the ConfigMap.",
+							Required:    true,
+							ForceNew:    true,
+						},
+						"namespace": {
+							Type:        schema.TypeString,
+							Description: "The namespace of the ConfigMap.",
+							Optional:    true,
+							ForceNew:    true,
+						},
+					},
+				},
+			},
+			"data": {
+				Type:        schema.TypeMap,
+				Description: "The data we want to add to the ConfigMap.",
+				Required:    true,
+			},
+			"force": {
+				Type:        schema.TypeBool,
+				Description: "Force overwriting data that is managed outside of Terraform.",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceKubernetesConfigMapV1DataCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	d.SetId(buildId(metadata))
```

Comment:
The function `buildId` sets ID as `metadata.namespace/metadata.name`, however, in this case, if `namespace` is omitted as optional value the ID will be set to `/metadata.name`. Can we make "default" a default value in the schema `metadata.namespace`? If that works, then we can rid of conditions that check whether the `namespace` attribute is set up in read and update functions.

Reply:
Good call – I've updated the schema.